src/Wellcome.Dds/Wellcome.Dds.Common.Tests/DdsIdentifierTests.cs

```diff
@@ -0,0 +1,205 @@
+using System;
+using FluentAssertions;
+using Xunit;
+
+namespace Wellcome.Dds.Common.Tests
+{
+    public class DdsIdentifierTests
+    {
+        private const string BNumber = "b99977766";
+        private const string Volume = "b99977766_0003";
+        private const string Issue = "b99977766_0002_0005";
+        private const string BNumberSequence = "b99977766/123";
+        private const string Unknown = "2b99977766_0002_0005/123";
+
+        [Theory(Skip = "Verify this is expected behaviour")]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(" ")]
+        public void Ctor_Throws_IfNullOrWhitespaceValue(string value)
+        {
+            // Act
+            Action action = () => new DdsIdentifier(value);
+            
+            // Assert
+            action.Should().Throw<ArgumentNullException>();
+        }
+        
+        [Fact]
+        public void Ctor_Correct_BNumber()
+        {
+            // Act
+            var identifier = new DdsIdentifier(BNumber);
+            
+            // Assert
+            identifier.IdentifierType.Should().Be(IdentifierType.BNumber);
+            identifier.BNumber.Should().Be("b99977766");
+            identifier.VolumePart.Should().BeNull();
+            identifier.IssuePart.Should().BeNull();
+            identifier.SequenceIndex.Should().Be(0);
+        }
+        
+        [Fact]
+        public void Ctor_Correct_Volume()
+        {
+            // Act
+            var identifier = new DdsIdentifier(Volume);
+            
+            // Assert
+            identifier.IdentifierType.Should().Be(IdentifierType.Volume);
+            identifier.BNumber.Should().Be("b99977766");
+            identifier.VolumePart.Should().Be("b99977766_0003");
+            identifier.IssuePart.Should().BeNull();
+            identifier.SequenceIndex.Should().Be(0);
+        }
+        
+        [Fact]
+        public void Ctor_Correct_BNumberAndSequenceIndex()
+        {
+            // Act
+            var identifier = new DdsIdentifier(BNumberSequence);
+            
+            // Assert
+            identifier.IdentifierType.Should().Be(IdentifierType.BNumberAndSequenceIndex);
+            identifier.BNumber.Should().Be("b99977766");
+            identifier.VolumePart.Should().Be("b99977766_123"); // NOTE: This doesn't seem right?

```

Comment:
yeah, that seems a stretch, but I hope we can retire this identifier form completely.
Leave the note there for now.